File: chanlib/src/main/java/com/mimireader/chanlib/models/ArchivedChanThread.java
package com.mimireader.chanlib.models;

import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class ArchivedChanThread extends ChanThread {
    @Expose
    private final String name;

    @Expose
    private final String domain;

    public ArchivedChanThread(@NonNull String boardName, long threadId, @NonNull List<ArchivedChanPost> posts, @NonNull String name, @NonNull String domain) {
        super(boardName, threadId, new ArrayList<ChanPost>(posts.size()));

        this.name = name;
        this.domain = domain;
        this.posts.addAll(posts);
    }

    public String getName() {
        return name;
    }

    public String getDomain() {
        return domain;
    }

    @NotNull
    @Override
    public String toString() {
        return "ArchivedChanThread{" +
                "board='" + boardName + '\'' +
                ", title='" + boardTitle + '\'' +
                ", thread id='" + threadId + '\'' +
                ", post count='" + posts.size() + '\'' +
                ", archive name='" + name + '\'' +
                ", archive domain='" + domain + '\'' +
                '}';
    }
}

File: chanlib/src/main/java/com/mimireader/chanlib/models/ArchivedChanThread.kt
package com.mimireader.chanlib.models

import com.google.gson.annotations.Expose
import java.util.*

class ArchivedChanThread(boardName: String, threadId: Long, posts: List<ArchivedChanPost>, @field:Expose val name: String?, @field:Expose val domain: String?) : ChanThread(boardName, threadId, ArrayList(posts.size)) {

    override fun toString(): String {
        return """ArchivedChanThread{board='$boardName', title='$boardTitle', thread id='$threadId', post count='${posts.size}', archive name='$name', archive domain='$domain'}"""
    }

    init {
        this.posts.addAll(posts)
    }
}